orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy

```diff
@@ -749,6 +762,19 @@ class TaskController {
     return pipelineConfigIds
   }
 
+  private static Boolean isNullOrEmpty(Object value) {
+    if (value == null) {
+      return true
+    } else if (value instanceof Collection) {
+      return value.isEmpty()
+    } else if (value instanceof String) {
+      return value.length() == 0
+    } else if (value == [:] || value == []) {
```

Comment:
Doesn't the `if instanceof Collection` cover `if value == []`?

Reply:
good catch yes. It just doesn't work for an empty map, which was somewhat surprising to me.